src/jgikbase/test/idmapping/util/util_test.py

```diff
@@ -1,28 +1,49 @@
-from jgikbase.idmapping.util.util import not_none
+from jgikbase.idmapping.util.util import not_none, check_string
 from jgikbase.test.idmapping.test_utils import assert_exception_correct
 from pytest import fail
+from jgikbase.idmapping.util import util
 
 
-def test_not_none_obj_pass():
+def test_not_none_pass():
     not_none(4, 'integer')
+    not_none('four', 'text')
 
 
-def test_not_none_string_pass():
-    not_none('four', 'text')
+def test_not_none_fail():
+    try:
+        not_none(None, 'my name')
+        fail('expected exception')
+    except Exception as got:
+        assert_exception_correct(got, ValueError('my name cannot be None'))
+
+
+def test_check_string_pass():
+    check_string('mystring', 'myname')
+    check_string('foo', 'bar', max_len=3)
+    check_string('foo', 'bar', legal_characters='fo')
+    check_string('foo', 'bar', 'fo', 3)
 
 
-def test_not_none_obj_fail():
-    fail_not_none(None, 'my name', ValueError('my name cannot be None'))
+def test_check_string_fail():
+    fail_check_string(None, 'foo', None, None, ValueError('foo cannot be None'))
```

Comment:
could add a test case where you pass a random type to check_string, such as a dict or custom object

Reply:
mypy will barf before the test can run